model/event/AbstractDacEvent.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ *
+ */
+
+namespace oat\taoDacSimple\model\event;
+
+use JsonSerializable;
+use oat\oatbox\event\Event;
+
+class AbstractDacEvent implements Event, JsonSerializable
```

Comment:
Very nice!

Reply:
if this is an abstract class probably it should be marked as `abstract` and named as DacEventAbstract?

Reply:
`abstract` - fixed. 
`AbstractClass` - named because of project code style and some naming conventions. What the reason of `ClassAbstract` ?

Reply:
Sorry. My bad. `AbstractClass` is perfect :)